cmd/soroban-cli/src/commands/env/mod.rs

```diff
@@ -0,0 +1,62 @@
+use crate::{
+    commands::global,
+    config::{
+        locator::{self},
+        Config,
+    },
+};
+use clap::Parser;
+
+#[derive(Debug, Parser)]
+pub struct Cmd {
+    #[command(flatten)]
+    pub config_locator: locator::Args,
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    Locator(#[from] locator::Error),
+
+    #[error("no defaults or environment variables set")]
+    NoEnv,
+}
+
+impl Cmd {
+    pub fn run(&self, _global_args: &global::Args) -> Result<(), Error> {
+        let config = Config::new()?;
+        let mut lines: Vec<(String, String)> = Vec::new();
+
+        if let Some(data) = get("STELLAR_NETWORK", config.defaults.network) {
+            lines.push(data);
+        }
+
+        if let Some(data) = get("STELLAR_ACCOUNT", config.defaults.identity) {
+            lines.push(data);
+        }
+
+        if lines.is_empty() {
+            return Err(Error::NoEnv);
+        }
+
+        let max_len = lines.iter().map(|l| l.0.len()).max().unwrap_or(0);
+
+        for (value, source) in lines {
+            println!("{value:max_len$} # {source}");
+        }
+
+        Ok(())
```

Comment:
This could all go in a `Display` impl for `Config`. Also is an empty config an error or just an empty string?

Reply:
Maybe we can use the rich printer to say that instead of failing, and returning an empty string. 🤔

Reply:
As for moving implementing this as a `Display` trait, I think it makes more sense to keep it here, because it also requires env vars, and the conditional to display the value. Wdyt?

Reply:
The printer is a good point. But I do think that `config::new()` maybe should use envs? Currently I guess we are relying on clap to handle it, but it could be handy for other plugins?

Reply:
> Currently I guess we are relying on clap to handle it, but it could be handy for other plugins?

That's fair, but I'd defer the work to a separate pr, because it's out of the scope of this one.